packages/mainsail/source/transaction.service.ts

```diff
@@ -209,15 +211,38 @@ export class TransactionService extends Services.AbstractTransactionService {
 	 * @musig
 	 */
 	public override async multiPayment(input: Services.MultiPaymentInput): Promise<Contracts.SignedTransactionData> {
-		return this.#createFromData("multiPayment", input, ({ transaction, data }) => {
-			for (const payment of data.payments) {
-				transaction.addPayment(payment.to, this.toSatoshi(payment.amount).toString());
-			}
+		if (!this.#isBooted) {
+			await this.#boot();
+		}
 
-			if (data.memo) {
-				transaction.vendorField(data.memo);
-			}
+		const transactionWallet = await this.clientService.wallet({
+			type: "address",
+			value: input.signatory.address(),
 		});
+
+		let builder = this.#app.resolve(MultiPaymentBuilder).nonce(transactionWallet.nonce().plus(1).toFixed(0));
+
+		if (input.fee) {
+			builder = builder.fee(this.toSatoshi(input.fee).toString());
+		}
+
+		if (input.data.memo) {
+			builder.vendorField(input.data.memo);
+		}
+
+		for (const { amount, to } of input.data.payments) {
+			builder = builder.addPayment(to, BigNumber.make(amount).toString());
```

Comment:
issue here is that if I pas `1` from arkvault it turns into `1` arktoshi. Will need a `this.toSatoshi` call first